cilium-cli: Test TLS with serverNames

jrajahalme opened the pull request:
Add new connectivity tests `client-egress-l7-tls-headers-sni` and `client-egress-tls-sni` to test SNI enforcement with and without TLS interception. Regressed patch releases are excluded from the new tests so that CI will not fail on downgrade tests.

```release-note
Add coverage for SNI enforcement in cilium-cli connectivity tests.
```

gandro commented on cilium-cli/cli/connectivity.go:
```diff
@@ -128,7 +128,7 @@ func newCmdConnectivityTest(hooks api.Hooks) *cobra.Command {
 	cmd.Flags().BoolVarP(&params.Timestamp, "timestamp", "t", false, "Show timestamp in messages")
 	cmd.Flags().BoolVarP(&params.PauseOnFail, "pause-on-fail", "p", false, "Pause execution on test failure")
-	cmd.Flags().StringVar(&params.ExternalTarget, "external-target", "one.one.one.one.", "Domain name to use as external target in connectivity tests")
+	cmd.Flags().StringVar(&params.ExternalTarget, "external-target", "one.one.one.one", "Domain name to use as external target in connectivity tests")
```
nit: Can't we sanitize this parameter in Cilium CLI? Seems error prone to require this from the user.

jrajahalme replied:
Good point! I'll add code to remove trailing dot.

jrajahalme replied:
Done!

gandro commented on cilium-cli/cli/connectivity.go:
```diff
@@ -58,6 +58,10 @@ func RunE(hooks api.Hooks) func(cmd *cobra.Command, args []string) error {
 
+		// Remove trailing dot from external-target paramater to make it compatible for
+		// use as an SNI
+		params.ExternalTarget, _ = strings.CutSuffix(params.ExternalTarget, ".")
```
nit: `strings.TrimSuffix` is the more appropriate helper here

jrajahalme replied:
Thanks, fixed!

marseel reviewed (approved):
FYI it seems like trailing dot was probably helping in CI: https://github.com/cilium/cilium/pull/31510
cc @giorio94

giorio94 reviewed (changes requested):
Requesting changes based on the comment above.

jrajahalme committed:
cilium-cli: Test TLS with serverNames

Add new connectivity tests client-egress-l7-tls-headers-sni and
client-egress-tls-sni to test SNI enforcement with and without TLS
interception.

Signed-off-by: Jarno Rajahalme <jarno@isovalent.com>

giorio94 reviewed (approved):
My previous concerns have been addressed, approving. Thanks!
